packages/adminrouter/extra/src/test-harness/tests/test_master.py

```diff
@@ -1,183 +1,367 @@
-    def test_redirect_req_without_slash(self, master_ar_process):
-        generic_no_slash_redirect_test(master_ar_process, '/exhibitor')
-
-    def test_if_exhibitor_endpoint_handles_redirects_properly(
-            self, master_ar_process, mocker, valid_user_header):
-        location_sent = 'http://127.0.0.1/exhibitor/v1/ui/index.html'
-        location_expected = 'http://127.0.0.1/exhibitor/exhibitor/v1/ui/index.html'
-        mocker.send_command(endpoint_id='http://127.0.0.1:8181',
-                            func_name='always_redirect',
-                            aux_data=location_sent)
-
-        url = master_ar_process.make_url_from_path("/exhibitor/v1/ui/index.html")
-        r = requests.get(url, allow_redirects=False, headers=valid_user_header)
-
-        assert r.status_code == 307
-        assert r.headers['Location'] == location_expected
-
-    def test_if_request_is_sent_to_correct_upstream(self,
-                                                    master_ar_process,
-                                                    valid_user_header):
-
-        generic_correct_upstream_dest_test(master_ar_process,
-                                           valid_user_header,
-                                           '/exhibitor/some/path',
-                                           'http://127.0.0.1:8181',
-                                           )
-
-    def test_if_upstream_request_is_correct(self,
-                                            master_ar_process,
-                                            valid_user_header):
-
-        generic_correct_upstream_request_test(master_ar_process,
-                                              valid_user_header,
-                                              '/exhibitor/some/path',
-                                              '/some/path',
-                                              )
-
-    def test_if_upstream_headers_are_correct(self,
-                                             master_ar_process,
-                                             valid_user_header):
-
-        generic_upstream_headers_verify_test(master_ar_process,
-                                             valid_user_header,
-                                             '/exhibitor/some/path',
-                                             )
-
-
-agent_prefix = '/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1'
+EndpointTestPaths = collections.namedtuple(
+    "EndpointTestPaths",
+    ("sent_path, expected_path")
+)
 
+EndpointTestConfig = collections.namedtuple(
+    "EndpointTestConfig",
+    ('test_paths,'
+     'jwt_forwarded, upstream_http_ver, upstream_headers, correct_upstream'),
+)
 
-class TestAgentEndpoint:
-    # FIXME: Figure out how we can test disable-request-response-buffering.conf
+# TODO: This is WIP/POC
+endpoint_test_configuration = [
+    EndpointTestConfig(
+        [EndpointTestPaths(
+            '/acs/api/v1/foo/bar', None),
+         ],
+        None, None, None, None),
```

Comment:
Maybe create some helper to avoid repeating `None` keyword. When working on `lynch` project @jgehrcke replaced `namedtuple` with `__slot__`.